baseplate/clients/redis.py

```diff
@@ -1,24 +1,56 @@
 from math import ceil
+from time import perf_counter
 from typing import Any
 from typing import Dict
 from typing import Optional
 
 import redis
 
-from prometheus_client import Gauge
-
 # redis.client.StrictPipeline was renamed to redis.client.Pipeline in version 3.0
 try:
     from redis.client import StrictPipeline as Pipeline  # type: ignore
 except ImportError:
     from redis.client import Pipeline
 
+from prometheus_client import Counter
+from prometheus_client import Gauge
+from prometheus_client import Histogram
+
 from baseplate import Span
 from baseplate.clients import ContextFactory
 from baseplate.lib import config
 from baseplate.lib import message_queue
 from baseplate.lib import metrics
 
+from baseplate.lib.prometheus_metrics import default_latency_buckets
+
+PROM_PREFIX = "redis_client"
+PROM_LABELS_PREFIX = "redis"
+
+PROM_SHARED_LABELS = [
+    f"{PROM_LABELS_PREFIX}_command",
+    f"{PROM_LABELS_PREFIX}_database",
+    f"{PROM_LABELS_PREFIX}_address",
```

Comment:
when this is set later, does the `host` include the port too?

Reply:
no, just the host, so for `"redis.url": "redis://localhost:1234/0"` we'll only get `localhost` because we call `self.connection_pool.connection_kwargs.get("host", "")`